api/components/docs/forms-and-redirects.md

```diff
@@ -0,0 +1,109 @@
+If the sanity checks fail, seller app will respond with `NACK` with an appropriate error code. Buyer app can retry with updated data.
+
+```mermaid
+sequenceDiagram
+    participant bap AS Distributor
+    participant bpp AS Form System
+    bap ->> bpp: submit the form on action url
+    alt submission successful
+        bpp ->> bap: sync response with `submission_id` and `status` as `SUBMITTED`
+    else submission failed
+        bpp ->> bap: sync response with `NACK`
+        bap ->> bpp: resubmit form w/ updated data
+        bpp ->> bap: sync response with `submission_id` and `status` as `SUBMITTED`
+    end
+```
+
+### `application/html` forms
+The buyer app redirects the user's browser to a seller app provided url. After completing the activity, seller app should take the responsibility to redirect the user's browser back to the buyer app so the user can continue his journey on the buyer application. We will follow the below conventions in handling these redirects.
+
+Seller app will redirect back the user to this url: `GET <bap_subscriber_url>/callback` with the following query params:
+1. `transaction_id` (transaction id used in the context)
+2. `status` - form status: `SUBMITTED` / `FAILED`
+3. `form_id`
+4. `submission_id` (only in success case)
+
+Seller app will additionally send `on_status` call containing the form status and `submission_id` in `form_response`
+
+```mermaid
+sequenceDiagram
+    participant bap AS Distributor
+    participant bppf AS Form System
+    participant bpp AS AMC/Aggregator
+    bap ->> bppf: redirect the user to form url
+    alt submission successful
+        bppf ->> bap: browser `/callback` with transaction id and success status
+        bpp ->> bap: `on_status` with `status` as `SUBMITTED` and `submission_id`
+    else submission failed
+        bppf ->> bap: browser `/callback` with transaction id and failure status
+        bpp ->> bap: `on_status` with `status` as `FAILED`
+        bap ->> bppf: redirect the user again to the same form url
+        bap ->> bppf: browser `/callback` with failure status (url can be used only once)
+        bpp ->> bap: `on_status` with `status` as `FAILED`
+    end
+```
+
+## Form Processing
+In both `text/html` and `application/html` forms: Buyer app will take `submission_id` received during form submission, use it in `form_response` in the next action call. Seller app will then process the form data
+
+#### if processing successful
+Seller app will either send the next form if applicable or send the `status` as `SUCCESS` in `form_response`.
+
+#### if processing failed
+Seller app will respond with error in the on action call. Buyer app can submit the same form with updated data, receive a submission id and make an action call for processing it again.
```

Comment:
if the business rules processing fails for the form during `select` call, should we send error in `on_select` call or use the `errors` array in `form_response`?

Reply:
business validation failure should be in `on_select` not in `errors` of `form_response`